go/storage/mkvs/db/badger/badger.go

```diff
@@ -271,6 +289,12 @@ func (d *badgerNodeDB) GetNode(root node.Root, ptr *node.Pointer) (node.Node, er
 
 	tx := d.db.NewTransactionAt(versionToTs(root.Version), false)
 	defer tx.Discard()
+
+	// Check if the root actually exists.
+	if err := d.checkRoot(tx, root); err != nil {
```

Comment:
Can you just quickly check what's the performance impact (since GetNode is in the hot path)? E.g. is it fine if we rely on Badger's memtables containing the root or would we benefit from a small LRU cache for roots?

Reply:
There doesn't seem to be an appreciable difference according to the storage benchmark debug command in the node.